### DISCUSSION TITLE
Redux Essentials documentation feedback

### DISCUSSION THREAD

**Discussion by jesse-willard-perficient-com** [reduxjs/redux#4079] [Ideas]
Some of the best documentation I have ever read, engaging and useful for learning React generally speaking

https://redux.js.org/tutorials/essentials/part-5-async-logic
**Async Logic and Data Fetching** - In this section, there is a code refactor that occurs about 2/3 of the way through, that is no indicated by the "changed code" highlighted lines that I think should be. **<PostExcerpt key={post.id} post={post} />** is extracted from the return to above the return if you jump ahead to the solution at the bottom. Instead, I think that refactor should be included on the step that introduces the PostExcerpt component. https://imgur.com/a/qnMmSAZ

**Comment by markerikson** [comment]
Yeah, there's several accumulated issues suggesting some tweaks for clarity along those lines: #4000, #4012 , https://github.com/reduxjs/redux-essentials-example-app/issues/1 .

Also, the async section really needs to be updated to show use of the "builder callback" syntax for `extraReducers` instead of the "object" form, since we want to promote the builder syntax as the default.

I will likely try to update this tutorial after we've gotten RTK 1.6 with RTK Query out the door, because I'll need to add a new section that demonstrates using RTK Query after all the existing content.

> Some of the best documentation I have ever read, engaging and useful for learning React generally speaking

And thank you! :)

**Reply by GabrielMolinar** [reply]
In the Essentials 5 async section I noticed something in the CodeSandbox at the end: when I click on "View Post" of one of the existing posts, I go to that post. Then when I refresh the page it says "Post not found!" @markerikson Hopefully you can address that while updating the async section?

That will really help me in my own project as well. And in general, Redux (using Toolkit) seems to be an excellent backbone for any MERN project! Thank you for that :)

**Reply by markerikson** [reply]
Yeah, it's a known bug in the example project code - see https://github.com/reduxjs/redux/issues/4212#issuecomment-961987815

**Comment by jesse-willard-perficient-com** [comment]
RE: features/posts/EditPostForm.js
In section 4 there is a brief mention to adding a route, which appears in section 6, other than that, I think section 4 is perfect.

Here is the code in section 6: https://imgur.com/a/ij19oeS

In section 6 the EditPostForm route appears as if it was already added to the file in a previous step, to help newbies with improving their workflow, it could be added immediately after the component definition in section 4. Perhaps adding the route directly after wouldn't be a bad idea. I like how the tutorial teaches proper workflow in all the previous sections and it make sense to add the route after creating the component file. People would just have it coded before section 6 and not think where it came from.

I suggest the following edit to section 4, which shouldn't take much time, but adds continuity to the work process:

{Directly under the EditPostForm component definition in section 4 (Using Redux Data)}

_Like with SinglePostPage, we'll need to import it into App.js and add a route that will render this component._
```
App.js
import { EditPostForm } from "./features/posts/EditPostForm";

<Route exact path="/posts/:postId" component={SinglePostPage} />
<Route exact path="/editPost/:postId" component={EditPostForm} />
<Redirect to="/" />
```
**We should also add a new link to our SinglePostPage that will route to EditPostPage, like:**

**Reply by markerikson** [reply]
Should be resolved in https://github.com/reduxjs/redux/pull/4153
